crates/frost/src/signer.rs

```diff
@@ -170,31 +180,27 @@ impl FrostSigner {
                     frost_secp256k1_tr::round2::sign(&request.signing_package, &nonces, &tweak_key_package)
                         .map_err(|e| SignerError::Internal(format!("Sign round2 failed: {e}")))?;
 
-                self.user_storage
-                    .set_user_state(
+                self.session_storage
+                    .set_session_state(
                         user_id.clone(),
-                        SignerUserState::SigningRound2 {
-                            tweak: tweak.clone(),
+                        session_id.clone(),
+                        SignerSessionState::SigningRound2 {
                             key_package: key_package.clone(),
-                            signature_share,
+                            tweak: tweak.clone(),
+                            signature_share: signature_share.clone(),
```

Comment:
```suggestion
                self.session_storage
                    .set_session_state(
user_id,
                        session_id.clone(),
                        SignerSessionState::SigningRound2 {
key_package,
                            tweak,
                            signature_share,
```

reduce clones
at the ending they aren't used, so we can just pass them as concrete value without cloning